bindings/java/src/org/sleuthkit/datamodel/OsAccountManager.java

```diff
@@ -727,6 +727,57 @@ void addOsAccountAttributes(OsAccount account, List<OsAccountAttribute> accountA
 
 	}
 	
+	/**
+	 * Get the OS account attributes for the given account.
+	 * 
+	 * @param account Account to get the attributes for.
+	 * 
+	 * @return List of attributes, may be an empty list.
+	 * 
+	 * @throws TskCoreException 
+	 */
+	List<OsAccountAttribute> getOsAccountAttributes(OsAccount account) throws TskCoreException {
+		
+		String queryString = "SELECT attributes.os_account_obj_id as os_account_obj_id, attributes.host_id as host_id, attributes.source_obj_id as source_obj_id, "
+				+ " attributes.attribute_type_id as attribute_type_id,  attributes.value_type as value_type, attributes.value_byte as value_byte, "
+				+ " attributes.value_text as value_text, attributes.value_int32 as value_int32, attributes.value_int64 as value_int64, attributes.value_double as value_double, "
+				+ " hosts.id, hosts.name as host_name, hosts.status as host_status "
+				+ " FROM tsk_os_account_attributes as attributes"
+				+ "		LEFT JOIN tsk_hosts as hosts "
+				+ " ON attributes.host_id = hosts.id "
+				+ " WHERE os_account_obj_id = " + account.getId();
+
+		try (CaseDbConnection connection = this.db.getConnection();
```

Comment:
Get the single user read lock before the connection

Reply:
Added a read lock.